bundles/core/org.eclipse.smarthome.core.persistence/src/main/java/org/eclipse/smarthome/core/persistence/internal/PersistenceServiceRegistryImpl.java

```diff
@@ -0,0 +1,101 @@
+/**
+ * Copyright (c) 2014-2016 by the respective copyright holders.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.core.persistence.internal;
+
+import java.net.URI;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Set;
+
+import org.eclipse.smarthome.config.core.ConfigOptionProvider;
+import org.eclipse.smarthome.config.core.ParameterOption;
+import org.eclipse.smarthome.core.persistence.PersistenceService;
+import org.eclipse.smarthome.core.persistence.PersistenceServiceRegistry;
+
+/**
+ * This is a central service for accessing {@link PersistenceService}s. It is registered through DS and also provides
+ * config options for the UI.
+ *
+ * @author Kai Kreuzer - Initial contribution and API
+ *
+ */
+public class PersistenceServiceRegistryImpl implements ConfigOptionProvider, PersistenceServiceRegistry {
+
+    private Map<String, PersistenceService> services = new HashMap<String, PersistenceService>();
+    private String defaultServiceId = null;
+
+    public PersistenceServiceRegistryImpl() {
+    }
+
+    public void addPersistenceService(PersistenceService service) {
+        services.put(service.getId(), service);
+    }
+
+    public void removePersistenceService(PersistenceService service) {
+        services.remove(service.getId());
+    }
+
+    protected void activate(Map<String, Object> config) {
+        modified(config);
+    }
+
+    protected void modified(Map<String, Object> config) {
+        if (config != null) {
+            this.defaultServiceId = (String) config.get("default");
```

Comment:
'this.' is not needed here. I personally prefer to remove it unless it's needed...